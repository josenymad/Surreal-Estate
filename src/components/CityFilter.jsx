import React from "react";
import { Link } from "react-router-dom";
import "../styles/city-filter.css";

const CityFilter = ({ setCity }) => {
  const cities = ["Manchester", "Leeds", "Sheffield", "Liverpool"];

  return (
    <div className="CityFilter">
      {cities.map((city) => {
        return (
          <Link
            key={city}
            className="city-filter__link"
            to={`/?query={"city": "${city}" }`}
            onClick={setCity(city)}
          >
            {city}
          </Link>
        );
      })}
    </div>
  );
};

export default CityFilter;
