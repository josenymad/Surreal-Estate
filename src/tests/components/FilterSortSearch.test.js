import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CityFilter from "../../components/FilterSortSearch";

describe("Filter, Sort, Search", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CityFilter />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 6 links", () => {
    render(
      <MemoryRouter>
        <CityFilter />
      </MemoryRouter>
    );

    const links = screen.getAllByRole("link");

    expect(links).toHaveLength(6);
  });
});
