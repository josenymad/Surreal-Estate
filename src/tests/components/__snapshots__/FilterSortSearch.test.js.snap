// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter, Sort, Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="filter-sort-search"
  >
    <div
      class="city-filter"
    >
      <p
        class="city-filter__label"
      >
        Filter by city
      </p>
      <a
        class="city-filter__link"
        href="/?query={\\"city\\":\\"Manchester\\"}"
      >
        Manchester
      </a>
      <a
        class="city-filter__link"
        href="/?query={\\"city\\":\\"Leeds\\"}"
      >
        Leeds
      </a>
      <a
        class="city-filter__link"
        href="/?query={\\"city\\":\\"Sheffield\\"}"
      >
        Sheffield
      </a>
      <a
        class="city-filter__link"
        href="/?query={\\"city\\":\\"Liverpool\\"}"
      >
        Liverpool
      </a>
    </div>
    <div
      class="price-sort"
    >
      <p
        class="price-sort__label"
      >
        Sort by price
      </p>
      <a
        class="price-sort__link"
        href="/?sort={\\"price\\":1}"
      >
        Low to High
      </a>
      <a
        class="price-sort__link"
        href="/?sort={\\"price\\":-1}"
      >
        High to Low
      </a>
    </div>
    <form
      class="search-form"
    >
      <label
        class="search-label"
        for="search"
      >
        Search 
      </label>
      <div
        class="search__input__button"
      >
        <input
          class="search-input"
          id="search"
          placeholder="bungalow"
        />
        <button
          class="search-button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
